org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelFormatterParticipant.java

```diff
@@ -67,4 +67,30 @@ public FormatElementCategory getFormatElementCategory(DOMElement element,
 		return null;
 	}
 
+	@Override
+	public boolean shouldCollapseEmptyElement(DOMElement element, SharedSettings sharedSettings) {
+		boolean enabled = sharedSettings.getFormattingSettings().isGrammarAwareFormatting();
+		if (!enabled) {
+			return true;
+		}
+		if (element.getAttribute("xsi:nil") != null) {
+			Collection<CMDocument> cmDocuments = contentModelManager.findCMDocument(element);
+			if (cmDocuments.isEmpty()) {
+				// The DOM document is not linked to a grammar, the collapse can be done.
+				return true;
+			}
+			for (CMDocument cmDocument : cmDocuments) {
+				CMElementDeclaration cmElement = cmDocument.findCMElement(element);
+				if (cmElement != null && cmElement.isEmpty()) {

```

Comment:
Why are you doing this test with isEmpty? Have you a test for that? Please adda comment which explains this condition.

Reply:
It was suggested in the comments here (https://github.com/eclipse/lemminx/issues/492#issuecomment-1253308411), I will add a test case for this

Reply:
This test case I had for this case still passes: `collapseEmptyElementsWithGrammarCanBeEmpty()`

Reply:
Ok I will write:

```java
if (cmElement != null) {
  if (cmElement.isEmpty()) {
    // TODO : add a sample
     return true;
  }
  if (!cmElement.isNillable()) {
    // TODO : add a sample
    return false;
  }
}
```

Reply:
I have removed this isEmpty() since it works well without it with the current logic